src/internal/nozzle/nozzle.go

```diff
@@ -343,20 +373,26 @@ func (n *Nozzle) createPointsFromGauge(envelope *loggregator_v2.Envelope) []*rpc
 	return points
 }
 
-func (n *Nozzle) createPointFromCounter(envelope *loggregator_v2.Envelope) *rpc.Point {
+func (n *Nozzle) createPointFromCounter(envelope *loggregator_v2.Envelope) []*rpc.Point {
+	tags := envelope.GetTags()
+	if !n.hasMatchedTags(tags) {
+		return []*rpc.Point{}
+	}
+
 	counter := envelope.GetCounter()
 	labels := map[string]string{
 		"source_id": envelope.GetSourceId(),
 	}
-	for k, v := range envelope.GetTags() {
+
+	for k, v := range tags {
 		labels[k] = v
 	}
-	return &rpc.Point{
+	return append([]*rpc.Point{}, &rpc.Point{
 		Timestamp: envelope.GetTimestamp(),
 		Name:      counter.GetName(),
 		Value:     float64(counter.GetTotal()),
 		Labels:    labels,
-	}
+	})
```

Comment:
Will this syntax work ? Without append call . return []*rpc.Point{
&rpc.Point{
Timestamp: envelope.GetTimestamp(),
Name:      counter.GetName(),
Value:     float64(counter.GetTotal()),
Labels:    labels,
},
}

Reply:
Yes this is also works well